runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunner.java

```diff
@@ -182,6 +183,13 @@ DataflowPipelineJob run(Pipeline pipeline, DataflowRunner runner) {
     return runner.apply(transform, input);
   }
 
+  /**
+   * Check success status of a dataflow job with given job metrics.
```

Comment:
This is not very accurate.

I think what this method is doing now is "checkForPAssertSuccess"

And, if we are not improving the code in this CL, could we add the following comment:
<p>If no PAssert is used, it will immediately return Optional.of(true) when metrics become available. But, the pipeline may fail later.